pkg/service.go

```diff
@@ -165,17 +165,23 @@ func (s *DefaultService) setReady() {
 }
 
 func (s *DefaultService) beaconEventLoop(ctx context.Context, client BeaconClient) error {
-	syncStatus, err := client.SyncStatus()
-	if err != nil {
-		return err
-	}
-	if syncStatus.IsSyncing {
-		return ErrBeaconNodeSyncing
-	}
+	for {
+		syncStatus, err := client.SyncStatus()
+		if err != nil {
+			return err
+		}
+		if syncStatus.IsSyncing {
```

Comment:
How difficult would it be to define an async `Synchronize(ctx context.Context) error` method on `BeaconClient`?  It would be much cleaner.

Reply:
I don't think the complexity is worth it in this case

Reply:
Okay, done